fairscale/optim/oss_utils.py

```diff
@@ -0,0 +1,107 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
```

Comment:
moving the different partitioning schemes to a dedicated file, to simplify a bit on the OSS side